core/yaml.adoc

```diff
@@ -0,0 +1,125 @@
+We also want this to be *idempotent*: that means that starting and restarting the server with the same command line should produce the same result and shouldn't fail.
+This feature is expected to be used as base for Bootable JAR implementation, it is not expected to be called directly by user (TODO get confirmation on this from Jeff and Brian and on how to 'hide' the feature).
+
+== Issue Metadata
+
+=== Issue
+
+* https://issues.redhat.com/browse/WFCORE[WFCORE-XXXX]
+
+=== Related Issues
+
+* https://issues.redhat.com/browse/WFLY[WFLY-XXXX]
+
+=== Dev Contacts
+
+* mailto:{email}[{author}]
+* mailto:jdenise@redhat.com[Jean-François Denise]
+
+=== QE Contacts
+
+=== Testing By
+// Put an x in the relevant field to indicate if testing will be done by Engineering or QE. 
+// Discuss with QE during the Kickoff state to decide this
+* [ ] Engineering
+
+* [ ] QE
+
+=== Affected Projects or Components
+
+=== Other Interested Projects
+
+=== Relevant Installation Types
+// Remove the x next to the relevant field if the feature in question is not relevant
+// to that kind of WildFly installation
+* [x] Traditional standalone server (unzipped or provisioned by Galleon)
+
+* [x] Managed domain
+
+* [x] OpenShift s2i
+
+* [x] Bootable jar
+
+== Requirements
+
+WildFly will be able to take a list of YAML files from the command line to redefine its exisiting configuration.
+The operations will be applied in the order the files are defined and after the initial operations defined by the XML configuration.
+The YAML configuration should be *idempotent* so that we can start and restart with the same command line without failure nor resulting with a different final configuration.
+The YAML structure should follow the management model structure. We may use YAML local tags to support operations syntax.
+So at boot, WildFly will load and parse its XML configuration, then add all the required extensions. It will then parse the YAML files, looking for a `wildfly-configuration` element as root fo the configuration updates.
+For each entry, it will check if the resource is already created by the XML, otherwise it will add a new operation to the list of boot operations to add the required resource.
+If the resource already exist then it will add a new operation to the list of boot operations to update the attributes.
+
+=== Hard Requirements
+
+* Support for multiple files.
+External files will take precedence over the XML configuration ones. Then order will follow the command line.
+
+* If the resource is already defined then all we have to du is updating its attrbiutes using the `write-attribute` operation.
+* If the YAML part define an empty resource and the resource already exists: log a WARNING, otherwise try to add it.
+* If the YAML is incorrect: like defining an non-existing attribute or child resource we should fail.
+
+=== Nice-to-Have Requirements
+
+* Undefine attribute: using a YAML tag !undefine. For example:
```

Comment:
off-topic but I'm looking at Helm Charts and their convention is to use `nil` to remove something (this could be applied both to resources and attributes here)

Reply:
But null is a supported value in some attributes.